### PR TITLE
add multi index with categories test

### PR DESCRIPTION
**Pull request by Hanspagh**
This PR adds test to address the issue in #49337, which have already been fixed on main

- [x] closses #49337
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Do you want a entry in the changelog since this is not a fix, just a test to ensure we dont run into this again?
<s>- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.<s>

### CODE REVIEW

File: pandas/tests/indexes/multi/test_setops.py [hunk 1]
```diff
@@ -14,9 +14,7 @@
 
 
 @pytest.mark.parametrize("case", [0.5, "xxx"])
-@pytest.mark.parametrize(
```
Comment (mroeschke) [inline pandas/tests/indexes/multi/test_setops.py:17]: Please avoid changing unrelated lines for style. We use black via precommit to autoformat the code base: https://pandas.pydata.org/pandas-docs/stable/development/contributing_codebase.html#pre-commit

Comment (Hanspagh) [reply pandas/tests/indexes/multi/test_setops.py]: Sorry about that, ran pre-commit to address

**Comment by Hanspagh** [comment]
Failing test seems unrelated to this

File: pandas/tests/indexes/multi/test_setops.py [hunk 2]
```diff
@@ -692,6 +692,43 @@ def test_intersection_lexsort_depth(levels1, levels2, codes1, codes2, names):
...
 
 
+@pytest.mark.parametrize(
+    "a",
+    [pd.Categorical(["a", "b"], categories=["a", "b"]), ["a", "b"]],
+)
+@pytest.mark.parametrize(
+    "b",
+    [
+        pd.Categorical(["a", "b"], categories=["b", "a"]),
```
Comment (mroeschke) [inline pandas/tests/indexes/multi/test_setops.py:702]: Is one of these supposed to have `ordered=True` like in the original issue? Otherwise both `b` params are the same

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Comment by Hanspagh** [comment]
rebased master and addresses the comment, I dont think the PR sync while it is closed so you cant see the newest commit

**Comment by mroeschke** [comment]
For some reason I cannot re-open this pull request. Mind opening up another one?
